File: 75-sort-colors/sort_colors.go
func sortColors(nums []int)  {
    // 大概算减治法
    low:=0
    // 分步计算, 遍历超过一次
    // 把所有0移到数组开头
    for p:=0;p < len(nums);p++{
        if nums[p] ==0 {
                nums[low],nums[p] = nums[p],nums[low]
            low ++
        }
    }
    // 把所有1移到数组开头
    for p:=low;p < len(nums);p++{
        if nums[p] ==1 {
                nums[low],nums[p] = nums[p],nums[low]
            low ++
        }
    }
    
}

File: 75-sort-colors/sort_colors.cc
class Solution {
public:
    void sortColors(vector<int>& nums) {
        unsigned low = 0;
        unsigned p = 0;
        int t;
        for(;p<nums.size();++p){
            if (nums[p] == 0){
                t = nums[p];
                nums[p] = nums[low];
                nums[low] = t;
                ++low;
            }
        }
        for(p = low;p<nums.size();++p){
            if (nums[p] == 1){
                t = nums[p];
                nums[p] = nums[low];
                nums[low] = t;
                ++low;
            }
        }
    }
};
